### PR TITLE
Scottx611x/galaxy rename dataset action

### LINKED ISSUE #2193: WorkflowTools need to be mindful of Galaxy's RenameDatasetAction
Currently as of [Release 1.6.0](https://github.com/refinery-platform/refinery-platform/commit/6003a4a7bb2f4591ee7042d606865837bd44499a), `WorkflowTool` launches do not take into account for when Galaxy Tools utilize `RenameDatasetActions` for their outputs.

I believe [createStepsCompact](https://github.com/refinery-platform/refinery-platform/blob/8380f09ca1d57069dbd96fb63b766a0a2a5fdf82/refinery/galaxy_connector/galaxy_workflow.py#L237) & [createStepsAnnot](https://github.com/refinery-platform/refinery-platform/blob/8380f09ca1d57069dbd96fb63b766a0a2a5fdf82/refinery/galaxy_connector/galaxy_workflow.py#L71) have examples of this.

---

- [x] Add tests

### PR DESCRIPTION
**Pull request by scottx611x**
Add support for Galaxy tool outputs that utilize [`RenameDatasetActions`](https://galaxyproject.org/learn/advanced-workflow/variables/#rename-output-based-on-specified-content)

Fix #2193

### CODE REVIEW

File: refinery/tool_manager/tests.py [hunk 1]
```diff
@@ -2108,6 +2108,33 @@ def test_attach_outputs_dataset_makes_proper_node_inheritance_chain(self):
 
         self._assert_analysis_node_connection_outputs_validity()
 
+    def test_that_galaxy_renamedatasetactions_are_handled(self):
```
Comment (hackdna) [inline refinery/tool_manager/tests.py]: `test_galaxy_renamedatasetaction_handling`?

File: refinery/tool_manager/tests.py [hunk 2]
```diff
@@ -2108,6 +2108,33 @@ def test_attach_outputs_dataset_makes_proper_node_inheritance_chain(self):
...
+                "output_name": "Refinery test tool LIST - N on data 4"
+            }
+        }
+        self.get_workflow_dict_mock.return_value = workflow_dict
+        self.galaxy_datasets_list_mock.start()
+
+        self.create_tool(ToolDefinition.WORKFLOW)
+        galaxy_datasets = self.tool._get_galaxy_history_dataset_list()
+        edited_galaxy_dataset = next(
+            (galaxy_dataset for galaxy_dataset in galaxy_datasets if
```
Comment (hackdna) [inline refinery/tool_manager/tests.py]: Unnecessary parentheses?

File: refinery/tool_manager/models.py [hunk 3]
```diff
@@ -1273,6 +1275,35 @@ def _update_galaxy_to_refinery_file_mapping_list(
...
+        )
+        if post_job_actions:
+            rename_dataset_key = (
+                "RenameDatasetAction{}".format(galaxy_dataset_dict["name"])
+            )
+            if rename_dataset_key in post_job_actions.keys():
+                galaxy_dataset_dict["name"] = (
+                    post_job_actions[
+                        rename_dataset_key
+                    ]["action_arguments"]["newname"]
```
Comment (mccalluc) [inline refinery/tool_manager/models.py]: perhaps `new_name`? Or is this matching something in galaxy?

Comment (scottx611x) [reply refinery/tool_manager/models.py]: Yeah this is a galaxy thing:/

File: refinery/tool_manager/tests.py [hunk 4]
```diff
@@ -2108,6 +2108,32 @@ def test_attach_outputs_dataset_makes_proper_node_inheritance_chain(self):
...
+        }
+        self.get_workflow_dict_mock.return_value = workflow_dict
+        self.galaxy_datasets_list_mock.start()
+
+        self.create_tool(ToolDefinition.WORKFLOW)
+        galaxy_datasets = self.tool._get_galaxy_history_dataset_list()
+        edited_galaxy_dataset = next(
+            galaxy_dataset for galaxy_dataset in galaxy_datasets if
+            self.tool._get_workflow_step(galaxy_dataset) == workflow_step
+        )
```
Comment (mccalluc) [inline refinery/tool_manager/tests.py]: I don't know that it's an interesting test, but what would you expect with just the comprehension? If you're not not really traversing the data structure, next is a little odd.

**Pull request merged**